internals/overlord/logstate/loki/loki_test.go

```diff
@@ -191,6 +191,65 @@ func (*suite) TestServerTimeout(c *C) {
 	c.Assert(err, ErrorMatches, ".*context deadline exceeded.*")
 }
 
+func (*suite) TestBufferFull(c *C) {
+	client := loki.NewClientWithOptions(
+		&plan.LogTarget{
+			Name:     "tgt1",
+			Location: "fake",
+		},
+		&loki.ClientOptions{
+			MaxRequestEntries: 3,
+		},
+	)
+
+	checkBuffer := func(expected []any) {
+		checkBuffer(c, client, expected)
+	}
+	addEntry := func(s string) {
+		err := client.Add(servicelog.Entry{Message: s})
+		c.Assert(err, IsNil)
+	}
+
+	checkBuffer([]any{nil, nil, nil, nil, nil, nil})
+	addEntry("1")
+	checkBuffer([]any{"1", nil, nil, nil, nil, nil})
+	addEntry("2")
+	checkBuffer([]any{"1", "2", nil, nil, nil, nil})
+	addEntry("3")
+	checkBuffer([]any{"1", "2", "3", nil, nil, nil})
+	addEntry("4")
+	checkBuffer([]any{nil, "2", "3", "4", nil, nil})
```

Comment:
Should we also be checking that `c.entries` is what we expect in each case? Something like:

```go
check([]any{nil, "2", "3", "4", nil, nil}, []string{"2", "3", "4"})
```